lib/tests/streamlit/runtime/state/query_params_test.py

```diff
@@ -0,0 +1,139 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+
+from streamlit.errors import StreamlitAPIException
+from streamlit.runtime.state.query_params import QueryParams
+from tests.delta_generator_test_case import DeltaGeneratorTestCase
+
+
+class QueryParamsMethodTests(DeltaGeneratorTestCase):
+    def setUp(self):
+        super().setUp()
+        self._query_params = QueryParams()
+        # avoid using ._query_params as that will use __setattr__
+        self._query_params.__dict__["_query_params"] = {"foo": "bar", "two": ["x", "y"]}
+
+    def test_getitem_nonexistent(self):
+        with pytest.raises(KeyError) as exception_message:
+            self._query_params["nonexistent"]
+            assert (
+                exception_message
+                == 'st.query_params has no key "nonexistent". Did you forget to initialize it?'
+            )
+
+    def test__getattr__nonexistent(self):
+        with pytest.raises(AttributeError) as exception_message:
+            self._query_params.nonexistent
+            assert (
+                exception_message
+                == 'st.query_params has no key "nonexistent". Did you forget to initialize it?'
+            )
+
+    def test_getitem_list(self):
+        print(self._query_params)
+        # get the last item in the array
+        assert self._query_params["two"] == "y"
+
+    def test_getitem_single_element_int(self):
+        # Internal state is always a string so we test setting it here
+        self._query_params["baz"] = 1
+        assert self._query_params["baz"] == "1"
+
+    def test_getitem_single_element_float(self):
+        # Internal state is always a string so we test setting it here
+        self._query_params["corge"] = 1.23
+        assert self._query_params["corge"] == "1.23"
+
+    def test__getitem__value(self):
+        assert self._query_params["foo"] == "bar"
+
+    def test_get(self):
+        assert self._query_params.get("foo") == "bar"
+
+    def test_get_default(self):
+        return_val = self._query_params.get("dog", default="bark")
+        assert return_val == "bark"
+
+    def test__getattr__(self):
+        assert self._query_params.foo == "bar"
+
+    def test__setitem__query_params(self):
+        assert "test" not in self._query_params
+        self._query_params["test"] = "test"
+        assert self._query_params.get("test") == "test"
+        message = self.get_message_from_queue(0)
+        assert (message.page_info_changed.query_string, "test=test")
```

Comment:
I don't think this assert actually does anything? It'd be equivalent to writing `assert (1, 2)`, which passes because `(1, 2)` is truthy

Reply:
Oops I copy and pasted `self.assertEquals` and then changed it to `assert`...